src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -646,6 +591,115 @@ void Compiler::optRelopImpliesRelop(RelopImplicationInfo* rii)
     }
 }
 
+//------------------------------------------------------------------------
+// optRelopTryInferWithOneEqualOperand: Given a domnating relop R(x, y) and
+// another relop R*(a, b) that share an operand, try to see if we can infer
+// something about R*(a, b).
+//
+// Arguments:
+//   domApp  - The dominating relop R*(x, y)
+//   treeApp - The dominated relop R*(a, b)
+//   rii     - [out] struct with relop implication information
+//
+// Returns:
+//   True if something was inferred; otherwise false.
+//
+bool Compiler::optRelopTryInferWithOneEqualOperand(const VNFuncApp&      domApp,
```

Comment:
If not being handled in this PR, could you log an issue for enabling the same to be done for TYP_SIMD?

We have all the same relops and same transforms that are valid there as well, so it'd be nice to ensure consistency

Reply:
I'm not really sure what that would entail. We already give up on relops between floating points because many of these inferences don't hold there.
I think you should open an issue about the inferences you think we should be able to make for these.

Reply:
Commutative swapping should still apply. That is, `x > y` is the same as `y < x` even for floating-point.

What doesn't hold true is that `x > y` is not the same as `!(x <= y)` (but only due to `NaN`, so if we know the inputs aren't `NaN` it becomes safe)

Reply:
I'll work on getting an issue open for it

Reply:
This PR doesn't do the kind of general canonicalization of VNs we were discussing on Discord the other day. Since @SingleAccretion pointed out that this came with non-trivial TP considerations, I just decided to do the easy thing and canonicalize right at this point, which is part of RBO's reasoning about inequalities. This PR doesn't really change much except it allows some of that reasoning to apply for a few more cases.